### PR TITLE
Render transaction info from XDR

### LINKED ISSUE #10: As a developer using simple signer, I want /sign to allow to receive an XDR via queryString (/sign?xdr=...) and render basic transaction details
**Background:**

While we sort out advanced integration with all websites, receiving an XDR via queryString shuld be possible, albeit discouraged since there are limits to the URL limit. In the future this project will use the iframe messaging approach to pass objects around instead of strings.

When the /sign page receives an xdr query parameter, it should display the transaction basic data, including: Source account, sequence number, date and time, validity (time bounds) and a simple list of operations in a rudimentary form (no need to cater for every special case, even displaying basic JSON details as text will do)

**Acceptance criteria:**

Please remember design is not important at this stage.

[ ] When `/sign` gets called with an `?xdr` query string, the XDR gets parsed and shown on screen as a transaction
[ ] If the XDR is invalid, a message shows up.

### PR DESCRIPTION
**Pull request by RivarolaNicolas**

### CODE REVIEW

File: cypress/integration/ui/sign.test.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const url = Cypress.env('HOST');
+it('alerts you if you do not pass an XDR as a query parameter or it is invalid', () => {
+    cy.visit(url);
+    cy.get('a[href*="sign"]').click();
+    cy.findByText('INVALID OR NULL XDR').should('exist');
+});
+it('renders transaction info if XDR query parameter is valid', () => {
+    cy.visit(`${url}/sign?xdr=${validXDR}`);
+    cy.get('.src-account').contains('GARFIPILHY3ZKN6XMDRRTQP3CRHDMXOEJO5CIC2JJNV4DM44DE5WYLOK');
+    cy.get('.operations').contains('GBFAIH5WKAJQ77NG6BZG7TGVGXHPX4SQLIJ7BENJMCVCZSUZPSISCLU5');
```
Comment (fsodano) [inline cypress/integration/ui/sign.test.ts]: const paymentRecipient = 'G...'

File: cypress/integration/ui/sign.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/// <reference types="cypress" />
+/// <reference types="@testing-library/cypress"/>
+
+const validXDR =
```
Comment (fsodano) [inline cypress/integration/ui/sign.test.ts]: Please put this in a fixture file which you can inmport, and call it for instance `paymentXdr`

File: cypress/integration/ui/sign.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    'AAAAAgAAAAAiVD0LPjeVN9dg4xnB+xRONl3ES7okC0lLa8GznBk7bACYloAACVO9AAAAAQAAAAEAAAAAAAAAAAAAAABh6rhSAAAAAAAAAAEAAAAAAAAACAAAAABKBB+2UBMP/abwcm/M1TXO+/JQWhPwkalgqizKmXyRIQAAAAAAAAABnBk7bAAAAEDacZu/NajCJ8BK05TjKL/YnsDDk+dbM3Zb8h9JRMGLZiWinI9bCnjal8T9SXU6hRAdFZSv12OZUtp4VwigwoUL';
+
+const url = Cypress.env('HOST');
+it('alerts you if you do not pass an XDR as a query parameter or it is invalid', () => {
+    cy.visit(url);
+    cy.get('a[href*="sign"]').click();
+    cy.findByText('INVALID OR NULL XDR').should('exist');
+});
+it('renders transaction info if XDR query parameter is valid', () => {
```
Comment (fsodano) [inline cypress/integration/ui/sign.test.ts]: Wrap multiple `it` conditions in a `describe` method that clusters them

File: package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
     "version": "0.0.1",
     "type": "module",
     "scripts": {
-        "dev": "vite",
+        "dev": "vite --open --https",
```
Comment (fsodano) [inline package.json]:
```suggestion
        "dev": "vite --https",
        "dev:open": "vite --https --open"
```

File: src/lib/Transaction.svelte [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<script lang="ts">
+    import { XDRIsValid } from './../store/state';
+    import StellarSdk from 'stellar-sdk';
+    let txData: any;
```
Comment (fsodano) [inline src/lib/Transaction.svelte]: this should be StellarSdk.Transaction

File: src/lib/Transaction.svelte [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            txData = new StellarSdk.Transaction(xdr, 'testnet');
+
+            return txData;
+        } catch (error) {
+            console.error(error);
+            return error;
+        }
+    };
+
+    getTxData(xdr);
```
Comment (fsodano) [inline src/lib/Transaction.svelte]:
```suggestion
            $isValidXdr= StellarSdk.xdr.TransactionEnvelope.validateXDR(xdr, 'base64');
            txData = new StellarSdk.Transaction(xdr, process.env.HORIZON_NETWORK_PASSPHRASE);
```

File: src/lib/Transaction.svelte [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+</script>
+
+{#if $XDRIsValid}
+    <div class="simple-signer payment-tx">
+        <p class="src-account">
+            Source account: {txData != undefined ? txData._source : ''}
```
Comment (fsodano) [inline src/lib/Transaction.svelte]:
```suggestion
            Source account: {txData ? txData._source : ''}
```

File: src/lib/Transaction.svelte [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        </p>
+        <p class="sequence-number">Sequence number: {txData != undefined ? txData._sequence : ''}</p>
+        <p class="time-bounds">
+            Time bounds: {txData != undefined
+                ? 'Min time ' + txData._timeBounds.minTime + ' Max time ' + txData._timeBounds.maxTime
```
Comment (fsodano) [inline src/lib/Transaction.svelte]: template literals

File: src/routes/Home.svelte [hunk 9]
```diff
@@ -0,0 +1,13 @@
+<script lang="ts">
+    import { Link } from 'svelte-navigator';
+</script>
+
+<h1>THIS IS THE HOME URL</h1>
```
Comment (fsodano) [inline src/routes/Home.svelte]:
```suggestion
<h1>Simple Signer</h1>
```

**Review by fsodano — CHANGES_REQUESTED** [review]
Please take a look at the comments, thank you!

File: src/store/state.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
+import { writable } from 'svelte/store';
```
Comment (fsodano) [inline src/store/state.ts]: this is not a store

File: cypress/integration/ui/sign.test.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+/// <reference types="cypress" />
+/// <reference types="@testing-library/cypress"/>
+import { paymentXdr, sourceAccount, paymentRecipient } from '../../fixtures/sign.json';
+
+const url = Cypress.env('HOST');
+describe('sign user history', () => {
```
Comment (fsodano) [inline cypress/integration/ui/sign.test.ts]: what history 🤔 ?

File: src/lib/Transaction.svelte [hunk 12]
```diff
@@ -0,0 +1,32 @@
+<script lang="ts">
+    import { writable } from 'svelte/store';
+    import * as StellarSdk from 'stellar-sdk';
+    import { Transaction } from 'stellar-sdk';
```
Comment (fsodano) [inline src/lib/Transaction.svelte]: I don't agree with this, it's very counter-intuitive

**Review by fsodano — CHANGES_REQUESTED** [review]

File: cypress/integration/ui/sign.test.ts (hunk 11 above)
Comment (RivarolaNicolas) [reply cypress/integration/ui/sign.test.ts]: I meant story, sorry. Anyway it's badly worded, I'm gonna give it a better name.

File: src/lib/Transaction.svelte [hunk 13]
```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+    import { writable } from 'svelte/store';
+    import * as StellarSdk from 'stellar-sdk';
+
+    const isValidXdr = writable(false);
+    let txData: StellarSdk.Transaction;
```
Comment (fsodano) [inline src/lib/Transaction.svelte]: simply `tx` is ok, everything is `data`

File: src/lib/Transaction.svelte [hunk 14]
```diff
@@ -0,0 +1,31 @@
+<script lang="ts">
+    import { writable } from 'svelte/store';
+    import * as StellarSdk from 'stellar-sdk';
```
Comment (fsodano) [inline src/lib/Transaction.svelte]: We discussed this and I said I prefer each dependency to be imported individually as to not obscure the actual dependencies. Import {Transaction, xdr} etc. from 'stellar-sdk'

**Pull request merged**